src/sentry/ownership/grammar.py

```diff
@@ -192,46 +216,46 @@ class Owner(namedtuple("Owner", "type identifier")):
         #team
     """
 
-    def dump(self):
+    def dump(self) -> Mapping[str, str]:
         return {"type": self.type, "identifier": self.identifier}
 
     @classmethod
-    def load(cls, data):
+    def load(cls, data: Mapping[str, str]) -> Owner:
         return cls(data["type"], data["identifier"])
 
 
-class OwnershipVisitor(NodeVisitor):
+class OwnershipVisitor(NodeVisitor):  # type: ignore
     visit_comment = visit_empty = lambda *a: None
 
-    def visit_ownership(self, node, children):
+    def visit_ownership(self, node: Any, children: Any) -> Sequence[Any]:
         return [_f for _f in children if _f]
 
-    def visit_line(self, node, children):
+    def visit_line(self, node: Any, children: Any) -> Any:
```

Comment:
I think we could probably type `node` and `children` on most of these functions, and then have the return values typed based on that. Having `Any` here kind of defeats the purpose of typing.

It might be easiest to just run `tests/sentry/ownership/test_grammar.py` and print out what values these can be. I'm pretty sure it's just be some `Node` container and similar for children.

Reply:
Did as much as I could here. This is kinda iffy since `children` is basically a variable length list and isn't consistent across method calls, i.e. depending on the method called, the list passed changes.